<?xml version="1.0"?>
<robot name="base_link">
<!-- Links -->
	<link name="base_link">
		<visual>
			<origin xyz="0 0 0.000005" rpy="0 0 0"/>
			<geometry>
				<cylinder radius="0.1" length="0.00001"/>
			</geometry>
			<material name="darkgray">
				<color rgba="0.5 0.5 0.5 0.0"/>
			</material>
		</visual>
	</link>
	<link name="intermediate">
		<visual>
			<origin xyz="0 0 0.000005" rpy="0 0 0"/>
			<geometry>
				<cylinder radius="0.1" length="0.00001"/>
			</geometry>
			<material name="darkgray">
				<color rgba="0.5 0.5 0.5 0.0"/>
			</material>
		</visual>
	</link>
	<link name="tcp">
		<visual>
			<origin xyz="0 0 0.05" rpy="0 0 0"/>
			<geometry>
				<cylinder radius="0.1" length="0.1"/>
			</geometry>
			<material name="magenta">
				<color rgba="1.0 0.0 1.0 1.0"/>
			</material>
		</visual>
	</link>

<!-- Links -->
	<joint name="x_joint" type="prismatic">
		<parent link="base_link"/>
		<child link="intermediate"/>
		<origin xyz="0 0 0"/>
		<axis xyz="1 0 0"/>
		<limit lower="-10.0" upper="10.0" effort="0.0" velocity="1.0"/>
	</joint>
	<joint name="y_joint" type="prismatic">
		<parent link="intermediate"/>
		<child link="tcp"/>
		<origin xyz="0 0 0"/>
		<axis xyz="0 1 0"/>
		<limit lower="-10.0" upper="10.0" effort="0.0" velocity="1.0"/>
	</joint>
</robot>
